lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -227,6 +227,8 @@ class DocumentImpl extends NodeImpl {
     }
 
     this._contentType = privateData.options.contentType;
+    this._encoding = privateData.options.encoding;
```

Comment:
I used "encoding" as the default encoding used for this document and others sub-documents and "characterSet" for the effective encoding. The names are maybe wrong, I can rename "encoding" as "defaultEncoding" and "characterSet" as "encoding".